internal/broker/broker_test.go

```diff
@@ -365,6 +367,7 @@ func TestIsAuthenticated(t *testing.T) {
 				"token": (&oauth2.Token{}).WithExtra(map[string]interface{}{"id_token": "invalid"}),
 			},
 		},
+		"Error when selected username does not match the provider one": {username: "not-matching", firstChallenge: "-", wantSecondCall: true},
```

Comment:
I won’t describe the arguments here, I would rather say "Error message returns valid json when error contains quotes" or something like that. As this is really what we want to test here and won’t remember this is the case in the future.

The issue is that the previous code could make the test pass too:
```            	            	data: '{"message": "could not get user info: could not fetch user info: returned user "test-user@email.com" does not match the selected one "not-matching""}'
```

I suggest we had, before creating the golden file, some validation in the tests that this is valid json. That way, the old code will really "fail" (and we don’t count on the developer to notice that the golden file can be or can’t be marshall to json, which is error-prone)

Reply:
I don't like the idea of having a test case that doesn't exactly match the function contract, but I agree with having JSON validation in the tests. I'll add checks for it on the returned values, wdyt?

Reply:
Pushed a commit that will fail the tests so you can see the outcome better: https://github.com/ubuntu/authd-oidc-brokers/pull/43/commits/d54e61d267338e124e8a6e1f14950654f0118ce4